Buildkite/ephemeral-agent.md

```diff
@@ -0,0 +1,107 @@
+# Using an ephemeral Buildkite agent to Run CI/CD Pipelines in Orka
+
+This guide explains how to set up an ephemeral Buildkite agent in a [Orka by MacStadium][orka] environment for your Buildkite builds. 
+
+The ephemeral agent is used to run a single Buildkite job in a Buildkite [pipeline][pipeline].  
+The agent is created prior to the job execution and deleted once the job completes.
+
+This is achieved by overwriting the Buildkite agent [bootstrap][bootstrap] command of the agent registered.  
+This way we create a wrapper agent that instead of executing the Buildkite job locally, spins up an Orka VM and delegates the job to that VM.
+
+## Requirements
+
+- [Orka][orka] VM
+- Buildkite [agent][agent]
+- [jq][jq] - command-line JSON processor used by the provided scripts
+
+## Set up an Orka VM base image
+
+The Orka VM used as an ephemeral agent needs to have the Buildkite agent installed locally. This is needed to allow the delegation of Buildkite jobs from the wrapper agent.
+
+Since the wrapper agents spins up a new Orka VM, the Buildkite agent must be included in the Orka VM base image.
```

Comment:
No, it should be a `VM`, not `agent`.
The wrapper can be any machine. During build, instead of running the build locally, it spins up a VM.
This VM has an agent as well, so the wrapper delegates the job to that agent via SSH.
Finally, the wrapper deletes the VM when the job is done